arbnode/resourcemanager/resource_management.go

```diff
@@ -90,18 +90,28 @@ type limitChecker interface {
 	String() string
 }
 
+func isSupported(c limitChecker) bool {
+	_, err := c.isLimitExceeded()
+	return err == nil
+}
+
 // newLimitChecker attempts to auto-discover the mechanism by which it
-// can check system limits. Currently Cgroups V1 is supported,
-// with Cgroups V2 likely to be implmemented next. If no supported
-// mechanism is discovered, it logs an error and fails open, ie
-// it creates a trivialLimitChecker that does no checks.
+// can check system limits. Currently Cgroups V1 and V2 are supported.
+// If no supported mechanism is discovered, it logs an error and
+// fails open, ie it creates a trivialLimitChecker that does no checks.
 func newLimitChecker(conf *Config) limitChecker {
```

Comment:
As I've mentioned in original review where this was introduced, please return concrete type here, as returning interfaces are not idiomatic in Go. You can change this function to:

func newLimitChecker(conf *Config) (cgroupsMemoryLimitChecker, error) {
       if c := newCgroupsMemoryLimitChecker(cgroupsV1MemoryFiles, conf.MemoryLimitPercent); isSupported(c) {
		return c, nil
	}
      if c := newCgroupsMemoryLimitChecker(cgroupsV2MemoryFiles, conf.MemoryLimitPercent); isSupported(c) {
            return c, nil
      }
      return nil, errNotSupported
}

Then in the call site you can call this, check with error.Is(err, errNotSupported) to instantiate trivialLimitChecker{} when needed.

Reply:
Refactored it as suggested.